app/models/profile.rb

```diff
@@ -15,6 +15,10 @@ class Profile < ApplicationRecord
     everyone: 4
   }, _prefix: :visible_to
 
+  scope :with_authenticated, -> { where(visibility: %i[everyone authenticated]) }
+  scope :nonblocked, ->(profile) { where.not(id: Friendship.blocks(profile).select(:buddy_id)) }
+  scope :befriended, ->(profile) { where(id: Friendship.friends_of(profile).select(:buddy_id), visibility: :friends) }
+
```

Comment:
I thought a lot about the names here, because some feel a little unintuitive, but ultimately decided that there aren't any noticeably better options. 

The only thing I'm considering is adjusting `befriended` to incorporate the higher level visibilities, which is based heavily on the conversation we had at RailsConf.

Right now, `befriended` only shows Profiles with friends visibility, vs all profiles that you've befriended.

To me that makes the scope very focused on visibility vs purely about friendship.

Reply:
Agree on the unintuitive part, but yeah, I also couldn't find anything clearer yet.

I also thought about wrapping the scopes up into something like `scope.connected_to(profile) || scope.visible_to_everyone`, but I wasn't sure about that.

Reply:
Sounds good on expanding the befriended scope to include everyone/authenticated viabilities.